### ISSUE TITLE
CanvasItem.draw_texture_rect_region parameter clip_uv doesn't have explanation

### ISSUE THREAD

**Issue by aXu-AP** [godotengine/godot-docs#8292]
**Your Godot version:**
4.2 beta 2

**Issue description:**
CanvasItem's `draw_texture_rect_region()` doesn't have any explanation as to what parameter `clip_uv` means.

**URL to the documentation page:**
https://docs.godotengine.org/en/stable/classes/class_canvasitem.html#class-canvasitem-method-draw-texture-rect-region

**Comment by AThousandShips** [comment]
Should also be added to `RenderingServer.canvas_item_add_texture_rect_region` and `Texture2D.draw_rect_region`

**Comment by aXu-AP** [comment]
As I was unsure what this parameter does and couldn't deduce by testing, I asked in rocket chat about it. I got explanation from clayjohn: 

> aXu-AP: Is `CanvasItem.draw_texture_rect_region` parameter `clip_uv` not working? I can't see any difference in output, I was expecting UV values to be clipped between 0...1 ie. stretched image. There's no word about in in documentation either, so I might be misintepreting it.

> clayjohn: It clips the UV to the source coordinates of the rect, so not exactly clipping to 0-1, just making sure you can't read from other elements in the same atlas I guess

As it stands I still don't understand it well enough to write to docs about it, but this could work as a starting point.

**Comment by k0T0z** [comment]
@aXu-AP You were working on it? If yes then I will close my PR in order for you to submit one, just bing me.

**Comment by aXu-AP** [comment]
@k0T0z No, I'm not working on it as I don't understand how it works. Even after clayjohn's and your explanation I either don't understand it or it doesn't work. Here's how I tested it:
```
func _draw() -> void:
	draw_texture_rect_region(texture, Rect2(0, 0, 256, 256), Rect2(0, 0, 32, 32), Color.WHITE, false, true)
```
And this is the result I get, both on `clip_uv = true` and `false`:
![kuva](https://github.com/godotengine/godot-docs/assets/1621768/6cc46701-e1fe-464f-be4e-762df9183c7b)
I expected the texture be either stretched out, or after reading your explanation on it, I expected that there wouldn't be bleeding from the other side of texture. Neither seems to be the case, so maybe I'm testing it wrong or still looking at wrong things.

**Comment by k0T0z** [comment]
@aXu-AP Can you send this project? I need to have a look, please.

**Comment by aXu-AP** [comment]
Sure. 
[ClipUV.zip](https://github.com/godotengine/godot-docs/files/13061197/ClipUV.zip)

**Comment by theraot** [comment]
Also `TileSet.uv_clipping`. I came here to report the missing documentation due to a question on GameDev StackExchange: [[Uv_clipping in godot?](https://gamedev.stackexchange.com/questions/208651/uv-clipping-in-godot)](https://gamedev.stackexchange.com/q/208651/10408), and found that this issue already exists.
